ResearchKit/ActiveTasks/ORKTowerOfHanoiViewController.m

```diff
@@ -0,0 +1,248 @@
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import "ORKTowerOfHanoiViewController.h"
+#import "ORKActiveStepViewController_Internal.h"
+#import "ORKTowerOfHanoiTowerView.h"
+#import "ORKActiveStepView.h"
+#import "ORKTowerOfHanoiTower.h"
+#import "ORKTowerOfHanoiStep.h"
+
+@interface ORKTowerOfHanoiViewController () <ORKTowerOfHanoiTowerViewDataSource, ORKTowerOfHanoiTowerViewDelegate>
+
+@end
+
+@implementation ORKTowerOfHanoiViewController {
+    ORKActiveStepCustomView *_towerOfHanoiCustomView;
+    NSNumber *_selectedIndex;
+    NSArray *_currentConstraints;
+    NSMutableArray *_towers;
+    NSArray *_towerViews;
+    NSInteger _numberOfDisks;
+    NSInteger _numberOfMoves;
+    NSTimer *_timer;
+    NSInteger _secondsElapsed;
+    NSDate *_firstMoveDate;
+}
+
+#pragma Mark -- UIViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    _towerOfHanoiCustomView = [ORKActiveStepCustomView new];
+    [_towerOfHanoiCustomView setTranslatesAutoresizingMaskIntoConstraints:NO];
+    self.activeStepView.activeCustomView = _towerOfHanoiCustomView;
+    _numberOfDisks = ((ORKTowerOfHanoiStep *)self.step).numberOfDisks;
+    
+    [self setupTowers];
+    [self setupTowerViews];
+    [self reloadData];
+    [self.activeStepView updateTitle:nil text: ORKLocalizedString(@"TOWER_OF_HANOI_TASK_ACTIVE_STEP_INTRO_TEXT", nil)];
+    [self setSkipButtonTitle:ORKLocalizedString(@"TOWER_OF_HANOI_TASK_ACTIVE_STEP_SKIP_BUTTON_TITLE", nil)];
+}
+
+- (void)viewWillDisappear:(BOOL)animated {
+    [super viewWillDisappear:animated];
+    [_timer invalidate];
+}
+
+- (void)updateViewConstraints {
+    [super updateViewConstraints];
+    if (_currentConstraints) {
+        [NSLayoutConstraint deactivateConstraints:_currentConstraints];
+        _currentConstraints = nil;
+    }
+    BOOL needCompactLayout = self.traitCollection.horizontalSizeClass == UIUserInterfaceSizeClassCompact &&
+    self.traitCollection.verticalSizeClass != UIUserInterfaceSizeClassCompact;
+    _currentConstraints = needCompactLayout ? [self compactConstraints] : [self regularConstraints];
+    [NSLayoutConstraint activateConstraints:_currentConstraints];
+}
+
+#pragma Mark -- ORKStepViewController
+
+- (void)skipForward {
+    [self finish];
+}
+
+#pragma Mark -- ORKActiveTaskViewController
+
+- (ORKResult *)result {
+    ORKTowerOfHanoiResult *result = [[ORKTowerOfHanoiResult alloc] initWithIdentifier:self.step.identifier];
+    result.numberOfMoves = _numberOfMoves;
+    result.puzzleWasSolved = [self puzzleIsSolved];
```

Comment:
Nice idea. I think perhaps the final configuration of all three towers would be useful if the problem is not solved. Do you agree?